ilastik/workflows/carving/opCarving.py

```diff
@@ -540,53 +506,32 @@ def saveCurrentObjectAs(self, name):
         self.AllObjectNames.meta.shape = (len(objects),)
 
         # now that 'name' is no longer part of the set of finished objects, rebuild the done overlay
-
         self._buildDone()
-        # self._clearLabels()
-        # self._mst.clearSegmentation()
-        # self.clearCurrentLabeling()
-        # self._mst.gridSegmentor.clearSeeds()
-        # self.Trigger.setDirty(slice(None))
-        # self.updatePreprocessing()
 
     def get_label_voxels(self):
         # the voxel coordinates of fg and bg labels
         if not self.opLabelArray.NonzeroBlocks.ready():
-            return (None, None)
-
-        nonzeroSlicings = self.opLabelArray.NonzeroBlocks[:].wait()[0]
-
-        coors1 = [[], [], []]
-        coors2 = [[], [], []]
-        for sl in nonzeroSlicings:
-            a = self.opLabelArray.Output[sl].wait()
-            w1 = numpy.where(a == 1)
-            w2 = numpy.where(a == 2)
-            w1 = [w1[i] + sl[i].start for i in range(1, 4)]
-            w2 = [w2[i] + sl[i].start for i in range(1, 4)]
+            return None, None
+
+        bg = [[], [], []]
+        fg = [[], [], []]
+        for slicing in self.opLabelArray.NonzeroBlocks[:].wait()[0]:
+            label = self.opLabelArray.Output[slicing].wait()
+            labels_bg = numpy.where(label == Labels.BACKGROUND)
+            labels_fg = numpy.where(label == Labels.FOREGROUND)
+            labels_bg = [labels_bg[d] + slicing[d].start for d in [1, 2, 3]]
+            labels_fg = [labels_fg[d] + slicing[d].start for d in [1, 2, 3]]
             for i in range(3):
-                coors1[i].append(w1[i])
-                coors2[i].append(w2[i])
+                bg[i].append(labels_bg[i])
+                fg[i].append(labels_fg[i])
 
         for i in range(3):
-            if len(coors1[i]) > 0:
-                coors1[i] = numpy.concatenate(coors1[i], 0)
-            else:
-                coors1[i] = numpy.ndarray((0,), numpy.int32)
-            if len(coors2[i]) > 0:
-                coors2[i] = numpy.concatenate(coors2[i], 0)
-            else:
-                coors2[i] = numpy.ndarray((0,), numpy.int32)
-        return (coors2, coors1)
+            bg[i] = numpy.concatenate(bg[i], 0) if len(bg[i]) > 0 else numpy.ndarray((0,), numpy.int32)
```

Comment:
It's a nitpick, but I think keyword arguments and standard `numpy.array` are a little bit nicer:
```suggestion
            bg[i] = numpy.concatenate(bg[i], axis=0) if len(bg[i]) > 0 else numpy.array((), dtype=numpy.int32)
```